### PR TITLE
Add a notebook on using ChatModel with Transformers and Pyfunc

### PR DESCRIPTION
**Pull request by daniellok-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/daniellok-db/mlflow/pull/11239?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11239/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11239
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

This PR adds some documentation for the new ChatModel pyfunc subclass:

1. A notebook tutorial showing how to easily build and serve an OpenAI-compatible chat model using the new `task` param in `mlflow.transformers`, and then how to add custom functionality by using `mlflow.pyfunc.ChatModel`.
2. A card in the New Features section
3. A card in the main Transformers index page

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [x] Yes. I've updated:
  - [x] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

A new notebook tutorial has been added on using the new `task` param in `mlflow.transformers`, and the new `mlflow.pyfunc.ChatModel` class.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/llms/transformers/tutorials/conversational/pyfunc-chat-model.ipynb [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    "\n",
+    "import mlflow\n",
+    "\n",
+    "generator = pipeline(\n",
+    "    \"text-generation\",\n",
+    "    model=\"TinyLlama/TinyLlama-1.1B-Chat-v1.0\",\n",
+    ")\n",
+    "\n",
+    "mlflow.transformers.save_model(\n",
+    "    transformers_model=generator, path=\"tinyllama-chat\", task=\"llm/v1/chat\"\n",
```
Comment (B-Step62) [inline docs/source/llms/transformers/tutorials/conversational/pyfunc-chat-model.ipynb:91]: [Not in the scope of this PR] Just realized that now an input example has to be string i.e. the one for original pipeline, not ChatModel. Created a JIRA in the next sprint:)

**Review by B-Step62 — COMMENTED** [review]
Left a few comments but overall looks great, thanks for working on this Daniel!

Btw are we planning to add documentation in `transformers/guide/index.html`? There are a few important notes e.g. supported task types, some param handling like `n`, `temperature`, which ideally users can find in docs rather than notebook.

**Comment by daniellok-db** [comment]
@B-Step62 Great points, I'll follow up! Yes I think we definitely want to add notes to the main transformers docs!

**Review by B-Step62 — APPROVED** [review]
Great demonstration for the cool feature, thanks!!

**Pull request merged**